### PR TITLE
hardening/397_implementation_details_doc

### PR DESCRIPTION
**Pull request by frbattid**
* Implements issue #397 
* No unit tests nor (unofficial) e2e tests are required
* Assignee @gtorodelvalle but LGTM from @fgalan is also required. Comments from @iariasleon are welcome as well

### CODE REVIEW

File: doc/design/OrionCKANSink.md [hunk 1]
```diff
@@ -206,10 +225,57 @@ A configuration example could be:
...
+A complete configuration as the described above is read from the given `Context` instance.
+
+[Top](#top)
+
+###<a name="section4.2"></a>`CKANBackendImpl` class
+This is a convenience backend class for CKAN that extends the `HttpBackend` abstract class (provides common logic for any Http connection-based backend) and implements the `CKANBackend` interface (provides the methods that any CKAN backend must implement). Relevant methods are:
+
+    public void persist(long recvTimeTs, String recvTime, String orgName, String pkgName, String resName, String attrName, String attrType, String attrValue, String attrMd) throws Exception;
+    
+Persists the context data regarding a single entity's attribute within the datastore associated to the given resource. This resource belongs to the given package/dataset, which in the end belongs to the given organization as well. Other notified attributes will be persisted by using this method, next to current one. This method creates the parts of the hierarchy (organization-pacakge/dataset-resource-datastore) if any of them is missing (row-like mode).
```
Comment (fgalan) [inline doc/design/OrionCKANSink.md]: pacakge -> package

File: doc/design/OrionSTHSink.md [hunk 2]
```diff
@@ -199,14 +215,55 @@ A configuration example could be:
...
+
+`MongoBackend` is created. This must be done at the `start()` method and not in the constructor since the invoking sequence is `OrionSTHSink()` (contructor), `configure()` and `start()`.
+
+    public void configure(Context);
+    
+A complete configuration as the described above is read from the given `Context` instance.
+
+[Top](#top)
+
+###<a name="section4.2"></a>`MongoBackend` class
```
Comment (fgalan) [inline doc/design/OrionSTHSink.md]: Is this the same class documented for OrionMongoSink? What a pitty Markdown doesn't allow include (http://stackoverflow.com/questions/4779582/markdown-and-including-multiple-files)...

Comment (frbattid) [inline doc/design/OrionSTHSink.md]: Yes , it is a pitty. Nevertheless, this comment leaded me to realize both `OrionMongoSink` and `OrionSTHSink` do not extend `OrionSink` but `OrionMongoBaseSink`. Fixing it...

File: doc/design/OrionCKANSink.md (hunk 1 above)
Comment (frbattid) [inline doc/design/OrionCKANSink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/96c960ccecc8c9d05818bc84d0b15516f3c03253

File: doc/design/OrionSTHSink.md (hunk 2 above)
Comment (frbattid) [inline doc/design/OrionSTHSink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/96c960ccecc8c9d05818bc84d0b15516f3c03253

**Comment by fgalan** [comment]
LGTM (passing the ball to @gtorodelvalle for final LGTM + merge + delete branch)

File: doc/design/OrionHDFSSink.md [hunk 3]
```diff
@@ -142,3 +212,5 @@ Fermín Galán Márquez (fermin.galanmarquez@telefonica.com) **[Contributor and
 Germán Toro del Valle (german.torodelvalle@telefonica.com) **[Contributor]**
 <br>
 Iván Arias León (ivan.ariasleon@telefonica.com) **[Quality Assurance]**
+
+[Top](#top)
```
Comment (gtorodelvalle) [inline doc/design/OrionHDFSSink.md]: Blank last line "requested" by Github :)

File: doc/design/OrionMongoSink.md [hunk 4]
```diff
@@ -118,3 +175,5 @@ Fermín Galán Márquez (fermin.galanmarquez@telefonica.com) **[Contributor and
 Germán Toro del Valle (german.torodelvalle@telefonica.com) **[Contributor]**
 <br>
 Iván Arias León (ivan.ariasleon@telefonica.com) **[Quality Assurance]**
+
+[Top](#top)
```
Comment (gtorodelvalle) [inline doc/design/OrionMongoSink.md]: Blank last line "requested" by Github :)

File: doc/design/OrionSTHSink.md [hunk 5]
```diff
@@ -199,14 +215,57 @@ A configuration example could be:
...
 <br>
 Fermín Galán Márquez (fermin.galanmarquez@telefonica.com) **[Contributor and Orion Context Broker owner]**
 <br>
 Germán Toro del Valle (german.torodelvalle@telefonica.com) **[Contributor]**
 <br>
-Iván Arias León (ivan.ariasleon@telefonica.com) **[Quality Assurance]**
\ No newline at end of file
+Iván Arias León (ivan.ariasleon@telefonica.com) **[Quality Assurance]**
+
+[Top](#top)
```
Comment (gtorodelvalle) [inline doc/design/OrionSTHSink.md]: Blank last line "requested" by Github :)

**Comment by gtorodelvalle** [comment]
Minor comments included ;)

Sadly, some rebasing is needed for merging :(

File: doc/design/OrionSTHSink.md (hunk 5 above)
Comment (frbattid) [inline doc/design/OrionSTHSink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/51d342c1565cf1b6efa79da68154b241c77cb4b7

File: doc/design/OrionMongoSink.md (hunk 4 above)
Comment (frbattid) [inline doc/design/OrionMongoSink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/51d342c1565cf1b6efa79da68154b241c77cb4b7

File: doc/design/OrionHDFSSink.md (hunk 3 above)
Comment (frbattid) [inline doc/design/OrionHDFSSink.md]: Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/51d342c1565cf1b6efa79da68154b241c77cb4b7

**Comment by gtorodelvalle** [comment]
LGTM! Thanks @frbattid ;)

**Pull request merged**